Solutions/Marain.ContentManagement.Api.Services/Marain/Cms/Api/Services/ContentSummaryService.cs

```diff
@@ -0,0 +1,73 @@
+﻿// <copyright file="ContentSummaryService.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Cms.Api.Services
+{
+    using System.Threading.Tasks;
+    using Marain.Cms.Api.Services.Internal;
+    using Menes;
+    using Menes.Hal;
+    using Microsoft.Net.Http.Headers;
+
+    /// <summary>
+    /// Handles request for content summary items.
+    /// </summary>
+    public class ContentSummaryService : IOpenApiService
+    {
+        /// <summary>
+        /// The ID for the getContentSummary operation.
+        /// </summary>
+        public const string GetContentSummaryOperationId = "getContentSummary";
+
+        private readonly ITenantedContentStoreFactory contentStoreFactory;
+        private readonly ContentSummaryResponseMapper contentSummaryMapper;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ContentService"/> class.
+        /// </summary>
+        /// <param name="contentStoreFactory">The content store for the service.</param>
+        /// <param name="contentSummaryMapper">The mapper for <see cref="ContentSummary"/> responses.</param>
+        public ContentSummaryService(
+            ITenantedContentStoreFactory contentStoreFactory,
+            ContentSummaryResponseMapper contentSummaryMapper)
+        {
+            this.contentStoreFactory = contentStoreFactory;
+            this.contentSummaryMapper = contentSummaryMapper;
+        }
+
+        /// <summary>
+        /// Get the content at the slug with the given ID.
+        /// </summary>
+        /// <param name="context">The context for the request.</param>
+        /// <param name="slug">The slug at which to create the content.</param>
+        /// <param name="contentId">The contentId for the content at the slug.</param>
+        /// <param name="ifNoneMatch">The value from the If-None-Match header, if provided.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        [OperationId(GetContentSummaryOperationId)]
+        public async Task<OpenApiResult> GetContentSummary(IOpenApiContext context, string slug, string contentId, string ifNoneMatch)
```

Comment:
Since, as it says, content is immutable, what is the ETag for?

Reply:
I suppose in a lot of cases clients won't know that it's immutable since they'll just be going via some HTTP facade, and we'll get an ETag because browsers cache things. There are two possible responses to that:

1. since its immutable, if you have a cached copy that must be right, so we just ignore the ETag and say "Yup, the version you have is good"
2. allow the possibility that there are bugs in HTTP stacks, and someone will present an ETag when they shouldn't have, and we should check it.

The issue that 2 deals with is particularly insidious when you have bad clients interacting with caching proxies. If a bad client sends an ETag when it shouldn't an intermediate cache may get the wrong idea about what the ETag should be and may persist that, which could then lead to things going weird when other, well-behaved clients try to fetch the same content.

But that's just a reverse-engineered excuse. I too am curious as to what the thinking was that lead to this.

Reply:
Your reverse engineering is a good argument - especially where symmetry is concerned.

Also, while our internal storage model is immutable and our current API doesn't provide for mutation, our ETag gives us a get-out clause if we choose to change the model further down the line...